components/style/properties/shorthand/font.mako.rs

```diff
@@ -97,14 +97,25 @@
     // This may be a bit off, unsure, possibly needs changes
     impl<'a> ToCss for LonghandsToSerialize<'a>  {
         fn to_css<W>(&self, dest: &mut W) -> fmt::Result where W: fmt::Write {
-            self.font_style.to_css(dest)?;
-            dest.write_str(" ")?;
-            self.font_variant.to_css(dest)?;
-            dest.write_str(" ")?;
-            self.font_weight.to_css(dest)?;
-            dest.write_str(" ")?;
-            self.font_stretch.to_css(dest)?;
+
+    % if product == "gecko":
+        % for name in "size_adjust kerning variant_caps variant_position".split():
+            if self.font_${name}.is_some() {
```

Comment:
I don't think this would build now. All subprops in `LonghandsToSerialize` are no longer `Option` after #15779, and basically they shouldn't be.

In this case, it should probably be something like
```rust
if self.font_${name} == font_${name}::get_initial_specified_value() {
    return Ok(());
}
```
but it is possible that not all of those longhands have `get_initial_specified_value()` method. In that case, you would need to add them. But most of them should be just single keyword properties, which would just work.